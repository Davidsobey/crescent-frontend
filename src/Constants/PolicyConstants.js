const PolicyConstants = {
  CREATE_REQUEST: 'CREATE_POLICY_REQUEST',
  CREATE_SUCCESS: 'CREATE_POLICY_SUCCESS',
  CREATE_FAILURE: 'CREATE_POLICY_FAILURE',

  CREATE_ACKNOWLEDGEMENT_REQUEST: 'CREATE_ACKNOWLEDGEMENT_REQUEST',
  CREATE_ACKNOWLEDGEMENT_SUCCESS: 'CREATE_ACKNOWLEDGEMENT_SUCCESS',
  CREATE_ACKNOWLEDGEMENT_FAILURE: 'CREATE_ACKNOWLEDGEMENT_FAILURE',

  ACKNOWLEDGE_REQUEST: 'ACKNOWLEDGE_REQUEST',
  ACKNOWLEDGE_SUCCESS: 'ACKNOWLEDGE_SUCCESS',
  ACKNOWLEDGE_FAILURE: 'ACKNOWLEDGE_FAILURE',

  GETALL_REQUEST: 'GETALL_POLICY_REQUEST',
  GETALL_SUCCESS: 'GETALL_POLICY_SUCCESS',
  GETALL_FAILURE: 'GETALL_POLICY_FAILURE',

  GETCLIENTPOLICIES_REQUEST: 'GETCLIENT_POLICIES_REQUEST',
  GETCLIENTPOLICIES_SUCCESS: 'GETCLIENT_POLICIES_SUCCESS',
  GETCLIENTPOLICIES_FAILURE: 'GETCLIENT_POLICIES_FAILURE',

  GETOUTSTANDINGPOLICIESFORUSER_REQUEST:
    'GETOUTSTANDING_POLICIES_FORUSER_REQUEST',
  GETOUTSTANDINGPOLICIESFORUSER_SUCCESS:
    'GETOUTSTANDING_POLICIES_FORUSER_SUCCESS',
  GETOUTSTANDINGPOLICIESFORUSER_FAILURE:
    'GETOUTSTANDING_POLICIES_FORUSER_FAILURE',
  GETOUTSTANDINGPOLICIESFORCLIENT_REQUEST:
    'GETOUTSTANDING_POLICIES_FORCLIENT_REQUEST',
  GETOUTSTANDINGPOLICIESFORCLIENT_SUCCESS:
    'GETOUTSTANDING_POLICIES_FORCLIENT_SUCCESS',
  GETOUTSTANDINGPOLICIESFORCLIENT_FAILURE:
    'GETOUTSTANDING_POLICIES_FORCLIENT_FAILURE',

  GETPOLICY_REQUEST: 'DELETE_POLICY_REQUEST',
  GETPOLICY_SUCCESS: 'DELETE_POLICY_SUCCESS',
  GETPOLICY_FAILURE: 'DELETE_POLICY_FAILURE',

  EDITPOLICY_REQUEST: 'EDIT_POLICY_REQUEST',
  EDITPOLICY_SUCCESS: 'EDIT_POLICY_SUCCESS',
  EDITPOLICY_FAILURE: 'EDIT_POLICY_FAILURE',

  DELETEPOLICY_REQUEST: 'DELETE_POLICY_REQUEST',
  DELETEPOLICY_SUCCESS: 'DELETE_POLICY_SUCCESS',
  DELETEPOLICY_FAILURE: 'DELETE_POLICY_FAILURE',

  CLEAR_POLICIES: 'CLEAR_POLICIES',

  LOAD_POLICY_REQUEST: 'LOAD_POLICY_REQUEST',
  LOAD_POLICY_SUCCESS: 'LOAD_POLICY_SUCCESS',
  LOAD_POLICY_FAILURE: 'LOAD_POLICY_FAILURE',

  UPLOAD_MATERIAL_REQUEST: 'UPLOAD_MATERIAL_REQUEST',
  UPLOAD_MATERIAL_SUCCESS: 'UPLOAD_MATERIAL_SUCCESS',
  UPLOAD_MATERIAL_FAILURE: 'UPLOAD_MATERIAL_FAILURE',

  GET_MATERIAL_REQUEST: 'GET_MATERIAL_REQUEST',
  GET_MATERIAL_SUCCESS: 'GET_MATERIAL_SUCCESS',
  GET_MATERIAL_FAILURE: 'GET_MATERIAL_FAILURE',

  GET_MATERIALS_REQUEST: 'GET_MATERIALS_REQUEST',
  GET_MATERIALS_SUCCESS: 'GET_MATERIALS_SUCCESS',
  GET_MATERIALS_FAILURE: 'GET_MATERIALS_FAILURE',

  LOAD_MATERIAL_REQUEST: 'LOAD_MATERIAL_REQUEST',
  LOAD_MATERIAL_SUCCESS: 'LOAD_MATERIAL_SUCCESS',
  LOAD_MATERIAL_FAILURE: 'LOAD_MATERIAL_FAILURE',
};

export default PolicyConstants;
