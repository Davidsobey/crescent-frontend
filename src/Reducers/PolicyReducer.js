import PolicyConstants from '../Constants/PolicyConstants';

function PolicyReducer(state = {}, action) {
  switch (action.type) {
    case PolicyConstants.CREATE_REQUEST:
      return {
        policies: action.policyName,
      };
    case PolicyConstants.CREATE_SUCCESS:
      return {
        policies: action.policyName,
      };
    case PolicyConstants.CREATE_FAILURE:
      return {};
    case PolicyConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case PolicyConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        policies: action.policies,
      });
    case PolicyConstants.GETALL_FAILURE:
      return {
        policies: {},
      };
    case PolicyConstants.GETOUTSTANDINGPOLICIESFORUSER_REQUEST:
      return {
        loading: true,
      };
    case PolicyConstants.GETOUTSTANDINGPOLICIESFORUSER_SUCCESS:
      return Object.assign({}, state, {
        policies: action.policies,
      });
    case PolicyConstants.GETOUTSTANDINGPOLICIESFORUSER_FAILURE:
      return {
        policies: {},
      };
    case PolicyConstants.CLEAR_POLICIES:
      return {
        policies: undefined,
      };
    default:
      return state;
  }
}

export default PolicyReducer;
