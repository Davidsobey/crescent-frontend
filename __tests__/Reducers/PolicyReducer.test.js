// Reducer to be tested
import PolicyReducer from '../../src/Reducers/PolicyReducer';
import PolicyConstants from '../../src/Constants/PolicyConstants';

describe('PolicyReducer', () => {
  test('CREATE_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.CREATE_SUCCESS, 
      newPolicyId: 1,
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });

  test('CREATE_ACKNOWLEDGEMENT_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.CREATE_ACKNOWLEDGEMENT_SUCCESS, 
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('ACKNOWLEDGE_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.ACKNOWLEDGE_SUCCESS, 
      policyName: 'ddd',
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('GETALL_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.GETALL_SUCCESS, 
      policies: [],
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('GETOUTSTANDINGPOLICIESFORUSER_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.GETOUTSTANDINGPOLICIESFORUSER_SUCCESS, 
      policies: [],
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('GETOUTSTANDINGPOLICIESFORCLIENT_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.GETOUTSTANDINGPOLICIESFORCLIENT_SUCCESS, 
      policies: [],
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('EDITPOLICY_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.EDITPOLICY_SUCCESS, 
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('DELETEPOLICY_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.DELETEPOLICY_SUCCESS, 
      id: 1,
    };

    const state = {
      policy: []
    }

    expect(PolicyReducer(state, action)).toMatchSnapshot();
  });
  
  test('LOAD_POLICY_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.LOAD_POLICY_SUCCESS, 
      policy: {id:1, name:'111'},
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('UPLOAD_MATERIAL_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.UPLOAD_MATERIAL_SUCCESS, 
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('GET_MATERIAL_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.GET_MATERIAL_SUCCESS, 
      material: {},
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
  
  test('GET_MATERIALS_SUCCESS', () => {
    const action = { 
      type: PolicyConstants.GET_MATERIALS_SUCCESS, 
      materials: [],
      policyId: 1,
      policyName: '111',
      policyDescription: '111',
      canAcknowlege: false,
    };

    expect(PolicyReducer(undefined, action)).toMatchSnapshot();
  });
});
