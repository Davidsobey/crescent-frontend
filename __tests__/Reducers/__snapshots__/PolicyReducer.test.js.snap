// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyReducer ACKNOWLEDGE_SUCCESS 1`] = `
Object {
  "acknowledging": false,
  "policy": "ddd",
}
`;

exports[`PolicyReducer CREATE_ACKNOWLEDGEMENT_SUCCESS 1`] = `
Object {
  "creating": false,
}
`;

exports[`PolicyReducer CREATE_SUCCESS 1`] = `
Object {
  "creating": false,
  "newPolicyId": 1,
}
`;

exports[`PolicyReducer DELETEPOLICY_SUCCESS 1`] = `
Object {
  "loading": false,
  "policy": Array [],
}
`;

exports[`PolicyReducer EDITPOLICY_SUCCESS 1`] = `
Object {
  "policy": undefined,
  "policy_editing": false,
}
`;

exports[`PolicyReducer GET_MATERIAL_SUCCESS 1`] = `
Object {
  "loadingMaterial": false,
  "policyMaterial": Object {},
}
`;

exports[`PolicyReducer GET_MATERIALS_SUCCESS 1`] = `
Object {
  "canAcknowlege": false,
  "policyDescription": "111",
  "policyId": 1,
  "policyMaterials": Array [],
  "policyMaterials_loading": false,
  "policyName": "111",
}
`;

exports[`PolicyReducer GETALL_SUCCESS 1`] = `
Object {
  "loading": false,
  "policies": Array [],
}
`;

exports[`PolicyReducer GETOUTSTANDINGPOLICIESFORCLIENT_SUCCESS 1`] = `
Object {
  "loading": false,
  "policyAcknowledgements": Array [],
}
`;

exports[`PolicyReducer GETOUTSTANDINGPOLICIESFORUSER_SUCCESS 1`] = `
Object {
  "loading": false,
  "policyAcknowledgements": Array [],
}
`;

exports[`PolicyReducer LOAD_POLICY_SUCCESS 1`] = `
Object {
  "policy": Object {
    "id": 1,
    "name": "111",
  },
  "policy_loading": false,
}
`;

exports[`PolicyReducer UPLOAD_MATERIAL_SUCCESS 1`] = `
Object {
  "creating": false,
}
`;
